pkg/inject/virtualgateway_envoy.go

```diff
@@ -111,6 +116,10 @@ func (m *virtualGatewayEnvoyConfig) mutate(pod *corev1.Pod) error {
 		pod.Spec.Containers[envoyIdx].ReadinessProbe = envoyReadinessProbe(m.mutatorConfig.readinessProbeInitialDelay,
 			m.mutatorConfig.readinessProbePeriod, strconv.Itoa(int(m.mutatorConfig.adminAccessPort)))
 	}
+
+	if m.mutatorConfig.enableSDS {
```

Comment:
Let's add a TODO to check for existing mounted `sds-socket-volume` in case the user has configured it already or alternatively skip it, as they have the ability to customize virtual gateway deployment spec unlike virtual nodes